assets/src/blocks/SocialMedia/SocialMediaEditorScript.js

```diff
@@ -0,0 +1,242 @@
+import { useEffect, useState } from '@wordpress/element';
+import { debounce } from 'lodash';
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+  RadioControl,
+  SelectControl,
+  PanelBody,
+} from '@wordpress/components';
+import { SocialMediaEmbed } from './SocialMediaEmbed';
+import { URLInput } from '../../components/URLInput/URLInput';
+import { HTMLSidebarHelp } from '../../components/HTMLSidebarHelp/HTMLSidebarHelp';
+import {
+  OEMBED_EMBED_TYPE,
+  FACEBOOK_EMBED_TYPE,
+  FACEBOOK_PAGE_TAB_TIMELINE,
+  FACEBOOK_PAGE_TAB_EVENTS,
+  FACEBOOK_PAGE_TAB_MESSAGES,
+  ALLOWED_OEMBED_PROVIDERS,
+} from './SocialMediaConstants.js';
+
+const { RichText } = wp.blockEditor;
+const { __ } = wp.i18n;
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+
+const loadScriptAsync = uri => {
+  return new Promise((resolve, reject) => {
+    let tag = document.createElement('script');
+    tag.src = uri;
+    tag.async = true;
+    tag.onload = () => {
+      resolve();
+    };
+    const body = document.getElementsByTagName('body')[0];
+    body.appendChild(tag);
+  });
+};
+
+const initializeTwitterEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.twttr) {
+      window.twttr.widgets.load();
+    }
+  }, 2000);
```

Comment:
Why does this need a timeout?

Reply:
I asked me the same, what if the end-user has a slow connection?

But until I don't feel completely familiar with the code base, I won't suggest significant changes.

Reply:
I'm not sure why this timeout is needed tbh, it was there before already 🤔 I think @sagarsdeshmukh initially worked on this?

Reply:
It's not a problem to leave the timeout in, also since it's only the editor and not the public JS, so not quite as performance sensitive context. But perhaps the conversion to hooks makes this easier to avoid? If it doesn't then we can maybe look at it later.

Reply:
> I'm not sure why this timeout is needed tbh, it was there before already 🤔 I think @sagarsdeshmukh initially worked on this?

This is the [initial commit](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/commit/43434bfacf5bfa5bab3ee4478ea265f830c29698), but not sure about it.